typed-racket-lib/typed-racket/rep/type-rep.rkt

```diff
@@ -579,6 +579,17 @@
   [#:frees (λ (f) (make-invariant (f value)))]
   [#:key 'continuation-mark-key])
 
+;; Newtype
+;; comes from define-new-subtype
+;; nm: a symbol representing the name of the Newtype
+;; id: a symbol created with gensym
+;; ty: a type for the representation
+(def-type Newtype ([nm symbol?] [id symbol?] [ty Type/c])
```

Comment:
My thinking is that since "subtype" means something outside of what a `Newtype` (or whatever you call it) is, we shouldn't call it `New-Subtype` because otherwise it would imply that new types have to be `New-Subtype` types to be subtypes of other types.